src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -73,10 +103,27 @@ static void MigrateBranches(Config config)
             // Map of current names and previous names
             var dict = new Dictionary<string, string[]>
             {
-                { "hotfix(es)?[/-]", new [] { "hotfix[/-]" }},
-                { "features?[/-]", new [] { "feature[/-]", "feature(s)?[/-]" }},
-                { "releases?[/-]", new [] { "release[/-]" }},
-                { "dev(elop)?(ment)?$", new [] { "develop" }}
+                {"hotfix(es)?[/-]", new[]
+                    {
+                        "hotfix[/-]"
+                    }
+                },
+                {"features?[/-]", new[]
+                    {
+                        "feature[/-]",
+                        "feature(s)?[/-]"
+                    }
+                },
+                {"releases?[/-]", new[]
+                    {
+                        "release[/-]"
+                    }
+                },
+                {"dev(elop)?(ment)?$", new[]
+                    {
+                        "develop"
+                    }
+                }
```

Comment:
done. 

I found Resharper rules that can do it, but they will affect all codebase. 

I think It will be better to extract method to make code itself cleaner and remove fear of Resharper clean code at all. 

```

            MigrateObsoleteBranches(config, "hotfix(es)?[/-]", "hotfix[/-]");
            MigrateObsoleteBranches(config, "features?[/-]", "feature[/-]", "feature(s)?[/-]");
            MigrateObsoleteBranches(config, "releases?[/-]", "release[/-]");
            MigrateObsoleteBranches(config, "dev(elop)?(ment)?$", "develop");
        }

        private static void MigrateObsoleteBranches(Config config, string newBranch, params string[] obsoleteBranches)
        {
            foreach (var branch in obsoleteBranches)
            {
                if (config.Branches.ContainsKey(branch))
                {
                    // found one, rename
                    var bc = config.Branches[branch];
                    config.Branches.Remove(branch);
                    config.Branches[newBranch] = bc; // re-add with new name
                }
            }
        }
```